pkg/controller/migcluster/validation.go

```diff
@@ -335,3 +345,44 @@ func (r *ReconcileMigCluster) validateSaTokenPrivileges(cluster *migapi.MigClust
 	}
 	return nil
 }
+
+// validate operator version.
+func (r ReconcileMigCluster) validateOperatorVersionMatchesHost(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	clusterConfig, err := cluster.GetClusterConfigMap(client)
+
+	operatorVersion, ok := clusterConfig.Data[migapi.OperatorVersionKey]
+	if !ok {
+		return errors.Errorf("configmap key not found: %v", migapi.OperatorVersionKey)
+
+	}
+
+	hostClusterConfig, err := cluster.GetClusterConfigMap(r)
+
+	hostOperatorVersion, ok := hostClusterConfig.Data[migapi.OperatorVersionKey]
+
+	operatorVersionMatchesHost := operatorVersion == hostOperatorVersion
+
+	if !operatorVersionMatchesHost {
+		cluster.Status.SetCondition(migapi.Condition{
+			Type:     OperatorVersionMismatch,
+			Status:   True,
+			Reason:   VersionCheckFailed,
+			Category: Warn,
+			Message:  fmt.Sprintf("This cluster is running a different version of the Migration Toolkit for Containers operator than the host cluster. Migrating to or from this cluster might result in a failed migration and data loss. Make sure all clusters are running the same version of the operator before attempting a migration."),
```

Comment:
+1, good error message.

Reply:
@ibolton336 is a `Warn` condition the right thing here?  I kind of expected `Critical` since migration will likely fail with mismatch, but don't recall what group consensus was.